tactics/tactics.ml

```diff
@@ -2250,24 +2250,25 @@ let one_constructor i lbind = constructor_tac false None i lbind
  *)
 
 let any_constructor with_evars tacopt =
-  let one_constr =
-    let tac i = constructor_tac with_evars None i NoBindings in
+  let one_constr ind =
+    let tac i = constructor_core with_evars (ind,i) NoBindings in
```

Comment:
Small nitpick. Here I had been careful not to create a closure over a closure in the most common case (`tacopt == None`). So I suggest moving the `ind` argument from `let one_constr` to both `let tac i` and `fun i`.

Reply:
I overlooked your intent, sorry. Change done.